src/utils/dnr-converter.js

```diff
@@ -9,131 +9,32 @@
  */
 
+import { convertWithAdguard } from '@ghostery/urlfilter2dnr';
 import { convert } from './dnr-converter-safari.js';
 
-let documentConverter;
-export function createDocumentConverter() {
-  const requestResolvers = new Map();
-
-  function createIframe() {
-    if (documentConverter) return documentConverter;
-
-    window.addEventListener('message', (event) => {
-      const requestId = event.data.rules.shift().condition.urlFilter;
-      let { rules, errors } = event.data;
-
-      const resolve = requestResolvers.get(requestId);
-
-      rules =
-        __PLATFORM__ === 'safari'
-          ? rules
-              .map((r) => {
-                try {
-                  return convert(r);
-                } catch (e) {
-                  errors.push(e);
-                }
-              })
-              .filter(Boolean)
-          : rules;
-
-      resolve({
-        rules,
-        errors: errors.map((e) => `DNR: ${e.message}`),
-      });
-
-      requestResolvers.delete(requestId);
-    });
-
-    const iframe = document.createElement('iframe');
-    iframe.setAttribute('src', 'https://ghostery.github.io/urlfilter2dnr/');
-    iframe.setAttribute('style', 'display: none;');
-
-    documentConverter = new Promise((resolve, reject) => {
-      iframe.addEventListener('load', () => resolve(iframe));
-      iframe.addEventListener('error', reject);
-
-      document.head.appendChild(iframe);
-    });
-
-    return documentConverter;
-  }
-
-  let requestCount = 0;
-
-  return async function convert(filter) {
-    const iframe = await createIframe();
-    const requestId = `request${requestCount++}`;
-
-    return new Promise((resolve) => {
-      requestResolvers.set(requestId, resolve);
-
-      iframe.contentWindow.postMessage(
-        {
-          action: 'convert',
-          converter: 'adguard',
-          filters: [requestId, filter],
-        },
-        '*',
-      );
-    });
-  };
+async function asyncConvert(filter) {
+  let { rules, errors } = await convertWithAdguard([filter]);
+
+  rules =
+    __PLATFORM__ === 'safari'
+      ? rules
+          .map((r) => {
+            try {
+              return convert(r);
+            } catch (e) {
+              errors.push(e);
+            }
+          })
+          .filter(Boolean)
+      : rules;
+
+  return { rules, errors };
 }
 
-async function setupOffscreenDocument() {
-  const path = 'pages/offscreen/urlfilter2dnr/index.html';
-  const offscreenUrl = chrome.runtime.getURL(path);
-  const existingContexts = await chrome.runtime.getContexts({
-    contextTypes: ['OFFSCREEN_DOCUMENT'],
-    documentUrls: [offscreenUrl],
-  });
-
-  if (existingContexts.length) {
-    return existingContexts[0];
-  }
-
-  await chrome.offscreen.createDocument({
-    url: path,
-    reasons: [chrome.offscreen.Reason.IFRAME_SCRIPTING],
-    justification: 'Convert network filters to DeclarativeNetRequest format.',
-  });
+export function createDocumentConverter() {
```

Comment:
This is pointless now, export `asyncConverter` (and rename it please to `convert` and rename Safari function to `convertToSafariFormat`), and use it directly where we used creators.